src/libraries/Common/src/System/Net/IPv6AddressHelper.Common.cs

```diff
@@ -2,17 +2,20 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 using System.Diagnostics;
+using System.Numerics;
 
-namespace System
+namespace System.Net
 {
     internal static partial class IPv6AddressHelper
     {
+        private const int Decimal = 10;
+        private const int Hex = 16;
         private const int NumberOfLabels = 8;
 
         // RFC 5952 Section 4.2.3
         // Longest consecutive sequence of zero segments, minimum 2.
         // On equal, first sequence wins. <-1, -1> for no compression.
-        internal static (int longestSequenceStart, int longestSequenceLength) FindCompressionRange(ReadOnlySpan<ushort> numbers)
+        internal static (int longestSequenceStart, int longestSequenceLength) FindCompressionRange(in ReadOnlySpan<ushort> numbers)
```

Comment:
`in` modifiers with Span parameters tend to be de-optimization on average (the affects depending on architecture).

Reply:
Thanks. That's a pity!

There's 3-5ns (of 40-25ns) of execution time which I can't account for, and applying the `in` modifier shaved around 8ns off my local benchmark; I had hoped that I'd just missed something obvious when optimising the IPv4 parsing path.

I'll remove them, but I don't know what other speed optimisations I can make without rewriting the path to use SIMD. 3-5ns is simultaneously a large percentage of `IPAddress.Parse`'s execution time, but also within the noise interval for the benchmarks on my laptop.

Reply:
If you have measured perf on multiple architectures and determined that these `in` modifiers help on average, you can certainly keep them. My comment was based on observation of similar attempts in the past.

Reply:
I haven't - I've been waiting for the build to finish to see if the improvements lined up between the Windows and the Linux JIT. I'll request an ARM64 benchmark too though